theano/tensor/var.py

```diff
@@ -533,7 +534,12 @@ def includes_bool(args_el):
         axis = None
         for i, arg in enumerate(args):
             try:
-                if arg is not np.newaxis:
+                if (isinstance(arg, (np.ndarray, theano.tensor.Variable)) and
```

Comment:
Why not modify Subtensor.convert() to also raise AdvancedIndexingError? This would make sure that elsewhere where we call convert(), if we forgot to update the code, we will end up using advanced indexing.